build-support/bin/ci.sh

```diff
@@ -105,17 +104,19 @@ esac
 # We're running against a Pants clone.
 export PANTS_DEV=1
 
-# Determine which Python interpreter to use for bootstrapping pex and for executing subprocesses.
-# Order matters here. We must constrain subprocesses before running the bootstrap stage,
-# or we will encounter the _Py_Dealloc error when bootstrapping a Python 3 PEX.
+# Note that we still must set PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS for Python 3, even though
+# ./pants3 does this already, because the constraints are only set when first bootstrapping the shard,
+# and will not apply for new shards. Without setting PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS, we
```

Comment:
This comment has become confusing again. It would be good to elaborate more clearly where interpreters are selected (creating the pex vs. running the pex vs subprocesses spawned by the pex) and how this setting affects that.